consensus/istanbul/backend/announce.go

```diff
@@ -589,130 +649,311 @@ func (sb *Backend) regossipAnnounce(msg *istanbul.Message, payload []byte) error
 	return nil
 }
 
-// ===============================================================
-//
-// define the AnnounceVersion messge format, the getAnnounceVersions send and handle function, and the announceVersions handle function
+// signedAnnounceVersion is a signed message from a validator indicating the most
+// recent version of its enode.
+type signedAnnounceVersion struct {
```

Comment:
Seeing as this is the same as `vet.SignedAnnounceVersionEntry`, one should be a type alias of the other. This will also eliminate athe need for the `newSignedAnnounceVersionFromEntry` function because you can just do a direct type cast.

Reply:
Holding off on this for the time being-- Kevin and I were thinking of storing the ecdsa public key in the db because it can be expensive to recover